build.gradle

```diff
@@ -24,132 +29,405 @@ if (verifySettingsGradle()) {
+                url 'https://maven.cleanroommc.com'
+            }
+        }
     }
     mavenLocal() // Must be last for caching to work
 }
 
-apply from: 'dependencies.gradle'
 
-//noinspection ConfigurationAvoidance
-for (File at : sourceSets.getByName('main').resources.files) {
-    if (at.name.toLowerCase().endsWith('_at.cfg')) {
-        tasks.deobfuscateMergedJarToSrg.accessTransformerFiles.from at
-        tasks.srgifyBinpatchedJar.accessTransformerFiles.from at
+// Dependencies
+
+// Configure dependency configurations
+configurations {
+    embed
+    implementation.extendsFrom(embed)
+}
+
+dependencies {
+    if (usesMixins.toBoolean() || forceEnableMixins.toBoolean()) {
+        implementation 'zone.rong:mixinbooter:7.0'
+        api('org.spongepowered:mixin:0.8.3') {
+            transitive = false
+        }
+        annotationProcessor('org.spongepowered:mixin:0.8.3') {
+            transitive = false
+        }
+
+        annotationProcessor 'org.ow2.asm:asm-debug-all:5.2'
+        // should use 24.1.1 but 30.0+ has a vulnerability fix
+        annotationProcessor 'com.google.guava:guava:30.0-jre'
+        // should use 2.8.6 but 2.8.9+ has a vulnerability fix
+        annotationProcessor 'com.google.code.gson:gson:2.8.9'
+    }
+
+    if (enableJUnit.toBoolean()) {
+        testImplementation 'org.junit.jupiter:junit-jupiter:5.9.1'
+        testImplementation 'org.hamcrest:hamcrest:2.2'
+    }
+
+    if (enableModernJavaSyntax.toBoolean()) {
+        annotationProcessor 'com.github.bsideup.jabel:jabel-javac-plugin:1.0.0'
+        compileOnly('com.github.bsideup.jabel:jabel-javac-plugin:1.0.0') {
+            transitive = false
+        }
+        // workaround for https://github.com/bsideup/jabel/issues/174
+        annotationProcessor 'net.java.dev.jna:jna-platform:5.13.0'
+        // Allow jdk.unsupported classes like sun.misc.Unsafe, workaround for JDK-8206937 and fixes Forge crashes in tests.
+        patchedMinecraft 'me.eigenraven.java8unsupported:java-8-unsupported-shim:1.0.0'
+
+        // allow Jabel to work in tests
+        testAnnotationProcessor "com.github.bsideup.jabel:jabel-javac-plugin:1.0.0"
+        testCompileOnly("com.github.bsideup.jabel:jabel-javac-plugin:1.0.0") {
+            transitive = false // We only care about the 1 annotation class
+        }
+        testCompileOnly "me.eigenraven.java8unsupported:java-8-unsupported-shim:1.0.0"
+    }
+}
+
+if (getFile('dependencies.gradle').exists()) {
+    apply from: 'dependencies.gradle'
+}
+
+
+// Test configuration
+
+// Ensure tests have access to minecraft classes
+sourceSets {
+    test {
+        java {
+            compileClasspath += patchedMc.output + mcLauncher.output
+            runtimeClasspath += patchedMc.output + mcLauncher.output
+        }
     }
 }
 
+test {
+    // ensure tests are run with java8
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(8)
+    }.get()
+
```

Comment:
Buildscript Repo: Why ensure tests are run with Java 8, when we apply jabel to the tests in the above block?